src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+                this._animation_styles = Object.assign({}, {
+                    open:this._animation_styles?.show,
+                    close:this._animation_styles?.hide});
+            }
+        } else if(name == 'fashion') {
+            this._fashion_style = getResponsiveValues(
+                new_val, this.DEFAULT_VALUES.fashion);
+            let val = this._checkFashion();
+            this._changeFashionStyle(val);
+        } else if(name == "colsize")
+            this._column_size = parseAttrVal(new_val);
+    }
+    override render() {
+        if(this._internal_fashion == 'accordion') {
+            return html`
+<div id='accordion' @click=${this._titleClicked}>
+    <slot name='acc'></slot>
+</div>
+            `
+        } else {
+            let fashionColumn:any = {
+                display: 'grid',
+                'grid-template-columns': `
+                    ${this._column_size.title} ${this._column_size.body}`
+            }
+            return html`
+${this._animateStyles()}
+<div id="parent" class="parentDiv"
+    style=${this._internal_fashion=='column'?
+            styleMap(fashionColumn):''}>
+    <div id="title-tab" style=
+        "display:flex;
+         flex-direction:${this._internal_fashion=='column'?
+            'column':'row'}">
+        <slot name="tab-title" @click=${this._titleClicked}>
+        </slot>
+    </div>
+    <div id="body-tab">
+        <slot id="body-slot" name="tab-body"></slot>
+    </div>
+</div>
+            `
+        }
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+        this.renderRoot.addEventListener("slotchange", (e) => {
+            this._slotChangedEvent(e)});
+        window.addEventListener("resize", this._resizeListener);
+    }
+    override disconnectedCallback() {
+        window.removeEventListener("resize", this._resizeListener);
+    }
+    override firstUpdated() {
+        if(!this.animation)
+            this.animationDur = 0;
+        this._openActive();
+    }
+    private _slotChangedEvent(event:Event) {
+        if(!event.target)
+            return;
+        let nodes = (event.target as HTMLSlotElement).assignedElements();
+        let name = (event.target as HTMLSlotElement)?.name;
+        if(name == 'tab-body') {
+            for(let i of nodes)
+                if(!i.classList.contains(this.activecls))
+                    (i as HTMLElement).style.display = 'none';
+        } else if(name == 'acc') {
+            for(let i of nodes)
+                if(!i.children[1].classList.contains(this.activecls))
+                    (i.children[1] as HTMLElement).style.display = 'none';
+        }
+    }
+    private _openActive() {
+        let val = this.querySelectorAll("[slot=tab-title]");
+        for(let i = 0;i < val.length; i++) {
+            if(val[i].classList.contains(this.activecls)) {
+                this.show(i);
+                break;
+            }
+        }
+        if(!this._prev_ix)
```

Comment:
What if prev_ix = 0 in this case, the show will be unnecessarily called. Better to check if prev_ix is null in this case.